examples/shallow_water/thermal_williamson2.py

```diff
@@ -0,0 +1,89 @@
+from gusto import *
+from firedrake import (IcosahedralSphereMesh, SpatialCoordinate, sin, cos)
+
+# ----------------------------------------------------------------- #
```

Comment:
it would be good to include the reference and a brief explanation in the comments